### ISSUE TITLE
Pulsar Broker 2.11.0-SNAPSHOT maven dependency is broken

### ISSUE THREAD

**Issue by eolivelli** [apache/pulsar#15346]
**Describe the bug**
If I create a Maven project that depends on pulsar-broker, 2.11.0-SNAPSHOT, when I build it Maven enters an infinite loop while resolving the dependencies.

**To Reproduce**
1) Build current Pulsar master locally
2) create a Maven project with a dependency on pulsar-broker, 2.11.0-SNAPSHOT
3) build the project

**Comment by lhotari** [comment]
I ran into a maven infinite loop recently and the resolution was #15234.

**Comment by eolivelli** [comment]
thanks @lhotari 

Adding this to my pom.xml unblocked the situation.

```
      <dependency>
        <groupId>org.apache.pulsar</groupId>
        <artifactId>pulsar</artifactId>
        <version>${pulsar.version}</version>
        <type>pom</type>
        <scope>import</scope>
      </dependency>
```

But I believe we have to fix the problem, otherwise many users will fall into troubles while updating their code.
It is very common to depend on pulsar-broker at least for tests

**Comment by lhotari** [comment]
> But I believe we have to fix the problem, otherwise many users will fall into troubles while updating their code.

It's probably a Maven bug similar to https://issues.apache.org/jira/browse/MNG-5592 . Perhaps a workaround is the only suitable approach?

**Comment by eolivelli** [comment]
I am using Maven 3.8.5, the problem is still there.

we should git bisect the changes in master branch and find out the problem.
it will take much time...

**Comment by tisonkun** [comment]
This seems no longer relevant?
